runtime/lib/backend/null_qubit/NullQubit.hpp

```diff
@@ -170,17 +233,47 @@ struct NullQubit final : public Catalyst::Runtime::QuantumDevice {
                         const std::vector<QubitIdType> &controlled_wires = {},
                         const std::vector<bool> &controlled_values = {})
     {
+        if (this->track_resources_) {
+            std::string prefix = "";
+            std::string suffix = "";
+            if (!controlled_wires.empty()) {
+                if (controlled_wires.size() > 1) {
+                    prefix += std::to_string(controlled_wires.size());
+                }
+                prefix += "C(";
+                suffix += ")";
+            }
+            if (inverse) {
+                prefix += "Adj(";
+                suffix += ")";
+            }
+            resource_data_["num_gates"]++;
+            resource_data_[prefix + name + suffix]++;
+        }
     }
 
     /**
      * @brief Doesn't apply a given matrix directly to the state vector of a device.
      *
      */
     void MatrixOperation(const std::vector<std::complex<double>> &,
-                         const std::vector<QubitIdType> &, bool,
+                         const std::vector<QubitIdType> &, bool inverse,
                          const std::vector<QubitIdType> &controlled_wires = {},
                          const std::vector<bool> &controlled_values = {})
     {
+        if (this->track_resources_) {
+            resource_data_["num_gates"]++;
+
+            std::string op_name = "QubitUnitary";
```

Comment:
constexpr std::string_view ?

Reply:
Can't use here since we modify this string